public/components/query_compare/search_result/result_components/result_grid.tsx

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import {
+  EuiButtonIcon,
+  EuiLink,
+  EuiPanel,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiIcon,
+  EuiIconTip,
+} from '@elastic/eui';
+import _, { uniqueId } from 'lodash';
+
+import { IDocType, SearchResults, Document } from '../../../../types/index';
+import { DocumentRank } from '../../../../contexts/utils';
+import { useSearchRelevanceContext } from '../../../../contexts';
+
+import './result_grid.scss';
+
+interface ResultGridComponentProps {
+  comparedDocumentsRank: DocumentRank;
+  queryResult: SearchResults;
+  resultNumber: number;
+}
+
+export const ResultGridComponent = ({
+  comparedDocumentsRank,
+  queryResult,
+  resultNumber,
+}: ResultGridComponentProps) => {
+  const { selectedIndex1, selectedIndex2 } = useSearchRelevanceContext();
+
+  const getExpColapTd = () => {
+    return (
+      <td className="osdDocTableCell__toggleDetails" key={uniqueId('grid-td-')}>
+        <EuiButtonIcon className="euiButtonIcon euiButtonIcon--text" iconType="arrowLeft" />
+      </td>
+    );
+  };
+
+  const getDlTmpl = (doc: IDocType) => {
+    return (
+      <div className="truncate-by-height">
+        <span>
+          <dl className="source truncate-by-height">
+            {_.toPairs(doc).map((entry: string[]) => {
+              return (
+                <span key={uniqueId('grid-desc')}>
+                  <dt>{entry[0]}:</dt>
+                  <dd>
+                    <span>{_.isObject(entry[1]) ? JSON.stringify(entry[1]) : entry[1]}</span>
+                  </dd>
+                </span>
+              );
+            })}
+          </dl>
+        </span>
+      </div>
+    );
+  };
+
+  const getTdTmpl = (conf: { clsName: string; content: React.ReactDOM | string }) => {
+    const { clsName, content } = conf;
+    return (
+      <td key={uniqueId('datagrid-cell-')} className={clsName}>
+        {typeof content === 'boolean' ? String(content) : content}
+      </td>
+    );
+  };
+
+  const getRankComparison = (documentId: string, documentRank: number) => {
+    const comparedRank = comparedDocumentsRank[documentId];
+
+    // No match result in comparison set
+    if (typeof comparedRank !== 'number') {
+      return (
+        <EuiText>
+          <span
+            style={{ color: '#69707D', fontSize: '12px', fontWeight: '400', lineHeight: '21px' }}
+          >
+            Not applicable
+          </span>{' '}
+          <EuiIconTip
+            aria-label="IconTip"
+            size="m"
+            type="questionInCircle"
+            color="#343741"
```

Comment:
You have hard-coded colors here. Do we know if it works as expected with change of themes in Dashboards. 
Check for theme setting here: https://playground.opensearch.org/app/management/opensearch-dashboards/settings